### ISSUE TITLE
Discovery does not complete when it fails for an incorrectly declared xunit testcase

### ISSUE THREAD

**Issue by smadala** [xunit/xunit#1610]
_From @sbaid on September 6, 2017 5:40_

## Description
dotnet.exe process (child of devenv.exe->vstest.console.exe) does not go away after test discovery failure. Test Explorer is stuck and continues to show discovery progress bar.

## Steps to reproduce
1. git clone http://github.com/sbaid/samples
2. open samples\bugs\XunitCoreParamTest\XunitCoreParamTest.sln
3. Build the project with test explorer pinned.

```csharp
// incorrect test code
[InlineData("hi", 1, 2, 3)]
[Theory]
public void Test1(params int[] args)
{
}
```
```csharp
// correct test code
[InlineData("hi", 1, 2, 3)]
[Theory]
public void Test1(string val, params int[] args)
{
}
```
## Expected behavior
Discovery completes with error 

## Actual behavior
Discovery error is seen in test output pane. 
dotnet.exe process does not go away & locks test binary.
test explorer continues to show discovery progress & stuck.
Error seen in test output pane:
```
[9/6/2017 11:09:14 AM Error] [xUnit.net 00:00:00.5934748] XunitCoreParamTest: Error creating Visual Studio test case for : System.InvalidCastException: At least one element in the source array could not be cast down to the destination array type.
   at System.Array.Copy(Array sourceArray, Int32 sourceIndex, Array destinationArray, Int32 destinationIndex, Int32 length, Boolean reliable)
   at Xunit.Sdk.TypeUtility.ResolveMethodArguments(MethodBase testMethod, Object[] arguments)
   at Xunit.Sdk.TestMethodTestCase.Initialize()
   at Xunit.Sdk.XunitTestCase.Initialize()
   at Xunit.Sdk.TestMethodTestCase.get_DisplayName()
   at Xunit.Runner.VisualStudio.TestAdapter.VsDiscoverySink.CreateVsTestCase(String source, ITestFrameworkDiscoverer discoverer, ITestCase xunitTestCase, Boolean forceUniqueName, LoggerHelper logger, Boolean designMode)

```
## Environment
Microsoft Visual Studio Enterprise 2017 
Version 15.3.3
VisualStudio.15.Release/15.3.3+26730.12
OS: win 10 15063.540

_Copied from original issue: Microsoft/vstest#1044_

**Comment by smadala** [comment]
_From @sbaid on September 6, 2017 5:40_

https://developercommunity.visualstudio.com/content/problem/107757/-error-xunitnet-project-error-creating-visual-stud.html

**Comment by smadala** [comment]
_From @singhsarab on December 28, 2017 7:1_

This is an issue specific to Xunit adapter/framework. The adapter is not returning test run complete.

**Comment by smadala** [comment]
@bradwilson Xunit adapter not completing discovery, Can you please look into this?

**Comment by bradwilson** [comment]
This was shipped fixed in 2.4 beta 1. I'm not necessarily 100% pleased with some of the asymmetry in the fix, but it should at least stop misbehaving now.
